src/index.js

```diff
@@ -1,2 +1,64 @@
 /* eslint-disable react/react-in-jsx-scope, react/jsx-filename-extension */
 /* @jsx createElement */
+function createElement(tagName, props, ...children) {
+  const element = document.createElement(tagName);
+
+  Object.entries(props || {}).forEach(([key, value]) => {
+    element[key.toLowerCase()] = value;
+  });
+
+  children.flat().forEach((child) => {
+    if (child instanceof Node) {
+      element.appendChild(child);
+      return;
+    }
+    element.appendChild(document.createTextNode(child));
+  });
+
+  return element;
+}
+
+const counter = {
+  currentCount: 0,
+};
+
+const app = document.getElementById('app');
+
+function render() {
+  function handleClick() {
+    counter.currentCount += 1;
+    render();
+  }
+
+  function handleClickNumber(value) {
+    counter.currentCount = value;
+    render();
+  }
+
+  const element = (
+    <div id="hello" className="hi">
+      <p>Hello, world!</p>
+      <p>
+        <button type="button" onClick={() => handleClick()}>
```

Comment:
```suggestion
        <button type="button" onClick={handleClick}>
```

이렇게 코드를 작성해줄 수도 있어요.
기존 코드와는 어떤 차이가 있을까요?

Reply:
`onClick={handleClick}` 로 수정해서 실행해봤는데 동일하게 동작하는 것으로 보입니다.

바로 눈에 보이는 차이점은 우선 코드가 간결해졌다는 것이고
그 외에는 꽤 오랜 시간 고민해보고 찾아보고 있는데... 아직까지 이해는 못했습니다. 😢 
도움이 될 만한 키워드가 있을까요?

Reply:
즉시 실행 함수로 넣어주는 경우 버튼을 누르는 쪽에서 비즈니스 로직을 포함하려고 하기 때문에 코드가 상대적으로 어려워져요.
만약 클릭할 때마다 복잡한 연산을 하는 코드를 만든다면 어떻게 될까요?
`onClick` 부분은 최대한 단순해야 하고,  그러기 위해선 클릭 했을 때 호출될 함수를 분리해줘서 함수만 전달하는 형태로 넣어주면 좋죠.

Reply:
음...제가 아직 잘 이해를 못하고 있는 것 같네요 ㅠㅠ

'즉시 실행 함수'는 함수 전체를 그룹 연산자 `()` 로 감싸주는 형태니까
위 코드를 즉시 실행 함수로 작성했다면 `onClick={ ( () => handleClick() ) }`이라고 생각이 들어서요 ㅠㅠ

51번 라인의 `<button type="button" onClick={() => handleClickNumber(i)}>` 코드에서
`onClick={() => handleClickNumber(i)}` 도 즉시 실행 함수로 호출한 것으로 이해하면 될까요?